pkg/catalog/routes.go

```diff
@@ -261,7 +261,7 @@ func getTrafficPoliciesForService(mc *MeshCatalog, routePolicies map[trafficpoli
 
 	for _, trafficTargets := range mc.meshSpec.ListTrafficTargets() {
 		log.Debug().Msgf("Discovered TrafficTarget resource: %s/%s", trafficTargets.Namespace, trafficTargets.Name)
-		if trafficTargets.Spec.Rules == nil || len(trafficTargets.Spec.Rules) == 0 {
+		if !isValidTrafficTarget(trafficTargets) {
 			log.Error().Msgf("TrafficTarget %s/%s has no spec routes; Skipping...", trafficTargets.Namespace, trafficTargets.Name)
```

Comment:
Not related to this PR - but the `log.Error()` seems more like an Info or perhaps a Warn, because it was not OSM's fault that SMI setup is missing a reference...

Reply:
ah that's fair. will handle that as a follow up.